src/CommunityToolkit.Maui/Views/Popup/Popup.shared.cs

```diff
@@ -40,22 +42,31 @@ public partial class Popup : Element, IPopup, IWindowController, IPropertyPropag
 	/// </summary>
 	public static readonly BindableProperty HorizontalOptionsProperty = BindableProperty.Create(nameof(HorizontalOptions), typeof(LayoutAlignment), typeof(Popup), LayoutAlignment.Center);
 
+	/// <summary>
+	///  Backing BindableProperty for the <see cref="Style" /> property. 
+	/// </summary>
+	public static readonly BindableProperty StyleProperty = BindableProperty.Create(nameof(Style), typeof(Style), typeof(Popup), default(Style), propertyChanged: (bindable, oldValue, newValue) => ((Popup)bindable).mergedStyle.Style = (Style)newValue);
+
 	readonly WeakEventManager dismissWeakEventManager = new();
 	readonly WeakEventManager openedWeakEventManager = new();
 	readonly Lazy<PlatformConfigurationRegistry<Popup>> platformConfigurationRegistry;
+	readonly MergedStyle mergedStyle;
 
 	TaskCompletionSource popupDismissedTaskCompletionSource = new();
 	TaskCompletionSource<object?> resultTaskCompletionSource = new();
 	Window? window;
+	ResourceDictionary resources = new();
+	bool IResourcesProvider.IsResourcesCreated => resources is not null;
 
 	/// <summary>
 	/// Instantiates a new instance of <see cref="Popup"/>.
 	/// </summary>
 	public Popup()
 	{
 		platformConfigurationRegistry = new Lazy<PlatformConfigurationRegistry<Popup>>(() => new(this));
-
-		VerticalOptions = HorizontalOptions = LayoutAlignment.Center;
+		((IResourceDictionary)resources).ValuesChanged += OnResourcesChanged;
```

Comment:
I don't have the code open right now to inspect the .NET MAUI implementation (in case it uses WeakEventManager) but do we need to unsubscribe from this?

Reply:
I write the unsubscribe handling in the OnClosed method.

Reply:
@brminnick , Thank you for fixing this.
I made sure you fixed it.